### PR TITLE
6344: make new bourbon select compatible with old inline select and w…

### PR DESCRIPTION
**Pull request by transitorytammy**
…ork with first enabled option

### CODE REVIEW

File: addon/components/new-bourbon-select-field.js [hunk 1]
```diff
@@ -38,6 +38,11 @@ export default Component.extend(ClickHandlerMixin, {
 
   didInsertElement() {
     this._super(...arguments);
+
+    if (this.get('value') !== this.get('selection.value')) {
+      this.set('value', this.get('selection.value'))
```
Comment (lawrencegetfeedback) [inline addon/components/new-bourbon-select-field.js]: where does `selection` come from?

File: addon/components/new-bourbon-select-field.js [hunk 2]
```diff
@@ -119,32 +124,28 @@ export default Component.extend(ClickHandlerMixin, {
 
       let index = this.get('internalContent').indexOf(valueHolder);
 
+      if ((valueHolder && !valueHolder.enabled) || ((!this.get('prompt') && index === -1))) {
```
Comment (lawrencegetfeedback) [inline addon/components/new-bourbon-select-field.js]: I think the first `valueHolder` and `index === -1` are both checking for existence (or non-existence), right? Might want to be more consistent here, like maybe change the first `valueHolder` to `index !== -1`.

File: addon/components/new-bourbon-select-field.js [hunk 3]
```diff
@@ -119,32 +124,28 @@ export default Component.extend(ClickHandlerMixin, {
...
     },
 
     set(key, value) {
-      this.set(
-        'value',
-        this.get('internalContent')
-          .objectAt(value)
-          .get('value')
-      );
+      if (value !== -1) {
```
Comment (lawrencegetfeedback) [inline addon/components/new-bourbon-select-field.js]: when would this property `selectedIndex` be set to `-1`?

File: addon/components/new-bourbon-select-field.js (hunk 1 above)
Comment (transitorytammy) [reply addon/components/new-bourbon-select-field.js]: that is the selection from below.

File: addon/components/new-bourbon-select-field.js (hunk 2 above)
Comment (transitorytammy) [reply addon/components/new-bourbon-select-field.js]: that is a good call.  i can change that.

File: addon/components/new-bourbon-select-field.js (hunk 3 above)
Comment (transitorytammy) [reply addon/components/new-bourbon-select-field.js]: when there is a prompt passed and there is no value.

File: addon/components/new-bourbon-select-field.js (hunk 1 above)
Comment (lawrencegetfeedback) [reply addon/components/new-bourbon-select-field.js]: ah okay, so this covers the case where the option associated with the `value` is disabled when this component is rendered, and so we need to find the next enabled option, and update `value`?

Comment (transitorytammy) [reply addon/components/new-bourbon-select-field.js]: yup.

File: addon/components/new-bourbon-select-field.js (hunk 2 above)
Comment (transitorytammy) [reply addon/components/new-bourbon-select-field.js]: so...`valueHolder` coming back `undefined` which is different than the `index !== 1` check.  and if i take it out then `!valueHolder.enabled` throws an error. so i will add `index !== 1` in addition to what i already have.

**Comment by transitorytammy** [comment]
1 day in june

**Pull request merged**